src/Core/ObjectRegistry.php

```diff
@@ -375,10 +376,10 @@ public function unload(string $objectName)
     /**
      * Returns an array iterator.
      *
-     * @return \ArrayIterator
-     * @psalm-return \ArrayIterator<string, TObject>
+     * @return \Traversable
```

Comment:
```suggestion
     * @return \Traversable<string, TObject>
```
TypeResolver should able to handle this. It generates a Collection type.